### PR TITLE
[Update] `GeometryEditor` sample enhancements

### PR DESCRIPTION
**Pull request by zkline101**
## Description
Changes: 
- Renames the geometry editor sample from `Sketch on map` -> `Create and edit geometries`
- Add shape tool feature to the sample
- Add uniform scale toggle to the sample 

## Linked Issue(s)

- `swift/issues/4382`

## How To Test

The shape tool can be tested by clicking on "Shapes" in the main menu and the uniform scale toggle can be tested in the menu after clicking on a tool.

## Screenshots
Simulator Screenshot - iPhone 14 Pro Max - 2023-08-11 at 15 29 43
Simulator Screenshot - iPhone 14 Pro Max - 2023-08-11 at 15 30 05

### CODE REVIEW

File: Shared/Samples/Create and edit geometries/sketch-on-map-1.png
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/sketch-on-map-1.png:1]: Can you recapture the screenshots as the sample title changed?

File: Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift [hunk 1]
```diff
@@ -282,4 +333,14 @@ class GeometryEditorMenuModel: ObservableObject {
             fatalError("Unexpected geometry type")
         }
     }
+    
+    /// Start editing with the specified tool and geometry type.
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift:337]:
```suggestion
    /// Starts editing with the specified tool and geometry type.
```

File: Shared/Samples/Create and edit geometries/README.md
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/README.md:1]: Can you also take a look at "How it works" section in the README and revise the phrasing according to `common-samples/pull/3858`? Or even better, compare the README to the design and fix the discrepancies. Thanks!

**Review by yo1995 — COMMENTED** [review]
Mostly looking good!

File: Shared/Samples/Create and edit geometries/README.md [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
 
 1. Create an `GeometryEditor` and assign it to a map view with the `geometryEditor` view modifier.
 2. Set the tool of the geometry editor to the preferred tool.
 3. Use the `start(withType:)` method on the `GeometryEditor` to start interactively sketching on the map view.
 4. Use various methods and properties of the `GeometryEditor` to undo, redo, delete a selected element, clear the sketch, and cancel the sketch.
-5. Save a sketch as a `Graphic` to a `GraphicsOverlay` displayed on the map view.
+5. Edit a tool's `InteractionConfiguration` to set the `GeometryEditorScaleMode` to allow either uniform or stretch scale mode.
+6. Save a sketch as a `Graphic` to a `GraphicsOverlay` displayed on the map view.
 
 ## Relevant API
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/README.md:26]: Can you match the tags to the design, and update the metadata again? Thanks

> draw, edit, freehand, geometry editor, sketch, vertex

File: Shared/Samples/Create and edit geometries/README.metadata.json [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {
-    "category": "Maps",
-    "description": "Use the Geometry Editor to edit or sketch a new point, multipoint, line, or polygon geometry on a map.",
+    "category": "",
```
Comment (yo1995) [inline Shared/Samples/Create and edit geometries/README.metadata.json:2]:
```suggestion
    "category": "Maps",
```

**Review by yo1995 — DISMISSED** [review]

File: Shared/Samples/Create and edit geometries/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+A field worker can mark features of interest on a map using an appropriate geometry. Features such as sample or observation locations, fences or pipelines, and building footprints can be digitized using point, multipoint, polyline, and polygon geometry types. Polyline and polygon geometries can be created and edited using a vertex-based creation and editing tool (i.e. vertex locations specified explicitly via tapping), or using a freehand tool.
+
+## How to use the sample
+
+Tap the pencil button to choose a geometry editor tool. Begin interactively sketching on the map view. Tap the pencil button again for editing options.
+
+## How it works
+
+1. Create an `GeometryEditor` and assign it to a map view with the `geometryEditor` view modifier.
```
Comment (pgruenler) [inline Shared/Samples/Create and edit geometries/README.md:18]:
```suggestion
1. Create a `GeometryEditor` and assign it to a map view with the `geometryEditor` view modifier.
```

**Review by pgruenler — CHANGES_REQUESTED** [review]
Looks good, just one grammar change.

**Review by yo1995 — APPROVED** [review]

**Review by pgruenler — APPROVED** [review]

**Pull request merged**